lib/Dialect/FIRRTL/Transforms/ModuleInliner.cpp

```diff
@@ -441,6 +476,42 @@ class Inliner {
 };
 } // namespace
 
+/// If this operation or any child operation has a name, add the prefix to that
+/// operation's name.  If the operation has any inner symbols, make sure that
+/// these are unique in the namespace.
+void Inliner::rename(StringRef prefix, Operation *op,
+                     ModuleNamespace &moduleNamespace) {
+  llvm::TypeSwitch<Operation *>(op)
```

Comment:
Is it worth just checking for hasAttr("name") and ignore the op type? I guess it would be in the caller.  Then we wouldn't lose names of expressions on inlining.

Reply:
Probably, yes. I'll do this after this lands (as this patch is just code motion of this region).

Reply:
Changed to rename anything with a `<StringAttr>("name")` in f5567434275091628dac1ab80ab642332ff0360a. This is a great observation as it helps both future proof the inliner and lets us do funky stuff like renaming named things from unknown dialects.